src/escrows/LPCurveYearnV2Escrow.sol

```diff
@@ -73,31 +90,43 @@ contract DolaFraxBPEscrow {
     function pay(address recipient, uint amount) public {
         if (msg.sender != market) revert OnlyMarket();
         uint256 tokenBal = token.balanceOf(address(this));
-        uint256 missingAmount = amount > tokenBal ? amount - tokenBal : 0;
-        // If there are enough tokens in the escrow, transfer the amount
+
         if (tokenBal >= amount) {
             token.safeTransfer(recipient, amount);
-        } else if (stakedBalance >= missingAmount) {
-            // If there are enough staked tokens in convex, withdraw the amount from convex
-            stakedBalance -= missingAmount;
-            rewardPool.withdrawAndUnwrap(missingAmount, false);
-            token.safeTransfer(recipient, amount);
-        } else if (
-            YearnVaultV2Helper.collateralToAsset(
-                yearn,
-                yearn.balanceOf(address(this))
-            ) >= missingAmount
-        ) {
-            // If there are enough tokens in Yearn, withdraw the amount from Yearn
-            uint256 withdrawAmount = YearnVaultV2Helper.assetToCollateral(
+            return;
+        }
+
+        uint256 missingAmount = amount - tokenBal;
+        uint256 convexBalance = rewardPool.balanceOf(address(this));
+        if (convexBalance > 0 && missingAmount > 0) {
```

Comment:
the check on `missingAmount > 0` is redundant. if it were 0 , line would be unreachable due to early return above.